inc/console/migration/domainsplugintocorecommand.class.php

```diff
@@ -0,0 +1,733 @@
+               $has_errors = true;
+
+               $message = sprintf(
+                  $core_type === null ?
+                     __('Unable to add domain type %s.') :
+                     __('Unable to update domain type %s.'),
+                  $typ['name']
+               );
+               $this->outputImportError($message, $progress_bar);
+               return false;
+            }
+
+            $this->addElementToMapping(
+               'PluginDomainsDomaintype',
+               $typ['id'],
+               'DomainType',
+               $new_tid ?? $core_type
+            );
+         }
+
+         $progress_bar->finish();
+         $this->output->write(PHP_EOL);
+      } else {
+         $this->output->writeln(
+            '<comment>' . __('No domains types found.') . '</comment>',
+            OutputInterface::VERBOSITY_NORMAL
+         );
+      }
+
+      return !$has_errors;
+   }
+
+   /**
+    * Migrate domains
+    *
+    * @throws LogicException
+    *
+    * @return boolean
+    */
+   protected function importDomains() {
+      $has_errors = false;
+
+      $this->output->writeln(
+         '<comment>' . __('Importing domains...') . '</comment>',
+         OutputInterface::VERBOSITY_NORMAL
+      );
+
+      $domains_iterator = $this->db->request([
+         'FROM'   => 'glpi_plugin_domains_domains',
+         'ORDER'  => 'id ASC'
+      ]);
+
+      $core_domains = [];
+      $cored_iterator = $this->db->request([
+         'SELECT' => ['id', 'name'],
+         'FROM'   => Domain::getTable()
+      ]);
+      while ($row = $cored_iterator->next()) {
+         $core_domains[$row['name']] = $row['id'];
+      }
+
+      if ($domains_iterator->count()) {
+         $progress_bar = new ProgressBar($this->output, $domains_iterator->count());
+         $progress_bar->start();
+
+         foreach ($domains_iterator as $dom) {
+            $progress_bar->advance(1);
+            $core_dom = null;
+
+            if (isset($core_domains[$dom['name']])) {
+               $core_dom = $core_domains[$dom['name']];
+               $message = sprintf(
+                  __('Updating existing domain %s...'),
+                  $dom['name']
+               );
+            } else {
+               $message = sprintf(
+                  __('Importing domain %s...'),
+                  $dom['name']
+               );
+            }
+            $this->writelnOutputWithProgressBar(
+               $message,
+               $progress_bar,
+               OutputInterface::VERBOSITY_VERY_VERBOSE
+            );
+
+            $mapped_type = $this->getCorrespondingItem('PluginDomainsDomaintype', $dom['plugin_domains_domaintypes_id']);
+            if ($types_id === null) {
+               $message = sprintf(
+                  __('Unable to find mapping for type %s.'),
+                  $dom['plugin_domains_domaintypes_id']
+               );
+               $this->outputImportError($message, $progress_bar);
+               return false;
```

Comment:
I would not consider it as a blocking error, and I would only output it with a normal verbosity

```php
            $this->writelnOutputWithProgressBar(
               $message,
               $progress_bar
            );
```